### PR TITLE
Add alpha module enable & disable commands

### PR DESCRIPTION
**Pull request by lindnerby**
**Description**

Changes proposed in this pull request:

- Add cmd `kyma alpha enable module [name]`
- Add cmd `kyma alpha disable module [name]`
- Available flags for both are:
  `-c, --channel string` to identify the channel of the module to enable
  ` -k, --kyma-name string` for the name of the Kyma to use, where empty defaults to 'default-kyma'
  `-n, --namespace string` the namespace of the Kyma to use, where empty defaults to namespace 'kyma-system'
  ` -t, --timeout duration` to set timeout for the cmd execution in seconds
  ` -w, --wait` defaults to false, wait for the Kyma resource's state change and reports it

**Related issue(s)**
* https://github.com/kyma-project/cli/issues/1328

### CODE REVIEW

File: cmd/kyma/alpha/disable/module/module.go
Comment (ruanxin) [inline cmd/kyma/alpha/disable/module/module.go:64]: kyma by default deployed in `kyma-system` namespace https://github.com/kyma-project/cli/blob/main/cmd/kyma/alpha/deploy/cmd.go#L75, I think we should align with it.

File: cmd/kyma/alpha/disable/module/module.go
Comment (ruanxin) [inline cmd/kyma/alpha/disable/module/module.go:50]: Can we give an example here already?

File: cmd/kyma/alpha/disable/module/module.go
Comment (ruanxin) [inline cmd/kyma/alpha/disable/module/module.go:173]: Is the logic here just trying to remove one entry in the slice? Then I suggest change to [this way](https://stackoverflow.com/a/37335777), just record the index first, then append the slice before and after.

https://stackoverflow.com/a/37335777

File: cmd/kyma/alpha/enable/module/module.go
Comment (ruanxin) [inline cmd/kyma/alpha/enable/module/module.go:65]: same here for the namespace

File: cmd/kyma/alpha/enable/module/module.go
Comment (ruanxin) [inline cmd/kyma/alpha/enable/module/module.go:119]: The above get kyma and extract modules logic which shared between enable and disable command should be extracted and put in `internal`.

File: cmd/kyma/alpha/disable/module/module.go
Comment (lindnerby) [reply cmd/kyma/alpha/disable/module/module.go:64]: I think by default is `kcp-system` at the moment? Should `list module` be adapted as well?

Comment (ruanxin) [reply cmd/kyma/alpha/disable/module/module.go:64]: yes, I think so, we should make all of them same by default in `kyma-system`

Comment (jakobmoellersap) [reply cmd/kyma/alpha/disable/module/module.go:64]: I changed the default namespace to kyma-system in my last PR as this aligns it with the managed offering of kyma. `kyma-system` should be used for all kyma resources, but control-plane resources, e.g. lifecycle/module manager can still stay in kcp-system

File: cmd/kyma/alpha/enable/module/module.go
Comment (lindnerby) [reply cmd/kyma/alpha/enable/module/module.go]: True, internalised the shared logic, but am missing a good name for the internal struct. Any good ideas?

File: cmd/kyma/alpha/enable/module/module.go
Comment (jakobmoellersap) [inline cmd/kyma/alpha/enable/module/module.go:142]: After rebasing on latest main, then building and running

`kyma provision k3d`
`kyma alpha deploy`
`kubectl apply -f https://raw.githubusercontent.com/kyma-project/kyma/main/modules/alpha/moduletemplate-keda.yaml`
`kyma enable module keda`

I receive: Kyma.operator.kyma-project.io "default-kyma" is invalid: spec.modules[0].channel: Invalid value: "": spec.modules[0].channel in body should be at least 3 chars long

I Think you need to exclude this in case the channel name is not set. Also maybe provide better documentation to mention that if not specified the default channel will be used.

File: cmd/kyma/alpha/enable/module/module.go
Comment (jakobmoellersap) [inline cmd/kyma/alpha/enable/module/module.go:58]: for the list command we use the positional arg 0 as the kyma name. I think we should either change the list command to also have the `k` flag or to align the positional argument here. WDYT?

File: cmd/kyma/alpha/enable/module/module.go
Comment (jakobmoellersap) [inline cmd/kyma/alpha/enable/module/module.go:112]: `kyma alpha enable module keda -c regular -w` with `k apply -f https://raw.githubusercontent.com/kyma-project/kyma/main/modules/alpha/moduletemplate-keda.yaml` results in a wait as long as timeout (with a really weird error message: `Error: failed to watch resource Kyma for state 'Ready': Timeout reached while waiting for kymas`) even though there should be an error reported as soon as the template is in alpha channel and the resulting kyma is in Error State. Then it should log this error state. By introspecting the status otherwise people will have to introspect it themselves which is unintuitive.

Similarly I would expect more output if it succeeds. Also if the kyma is already in error state and you then run `kyma alpha enable module keda -c regular -w` it simply does an EXIT 0

**Review by ruanxin — APPROVED** [review]

File: cmd/kyma/alpha/disable/module/module.go
Comment (jakobmoellersap) [inline cmd/kyma/alpha/disable/module/module.go:38]: This output renders:

```

Examples:
Example:
Enable "my-module" from "alpha"" channel in "default-kyma" from "kyma-system" namespace
		kyma alpha enable module my-module -c alpha -n kyma-system -k default-kyma
```
so you should omit the `Example:`

**Review by jakobmoellersap — CHANGES_REQUESTED** [review]

File: cmd/kyma/alpha/enable/module/module.go
Comment (lindnerby) [reply cmd/kyma/alpha/enable/module/module.go:58]: Good idea, but I'm leaning towards having positional arg as module name, since kyma name and namespace can be omitted using defaults.
Also if introducing `-k` for `list` I will default it as well to "default-kyma", you agree?

File: internal/cli/defaults.go
Comment (lindnerby) [reply internal/cli/defaults.go:5]: Sure, done.

File: cmd/kyma/alpha/enable/module/module.go
Comment (lindnerby) [reply cmd/kyma/alpha/enable/module/module.go:142]: Thanks for the find. What is the default channel? alpha?
Also kymas `spec.modules` items can consist only of `name:` field? `channel` is optional right?

Comment (jakobmoellersap) [reply cmd/kyma/alpha/enable/module/module.go:142]: so from my view you do not need a channel at all. you can add the module and the channel of the kyma will be used. The problem is that you are using unstructured and you sent an empty channel, so leaving it out if its empty should be sufficient 👍

File: cmd/kyma/alpha/enable/module/module.go
Comment (lindnerby) [reply cmd/kyma/alpha/enable/module/module.go:114]: Valid point. Updated the watch mechanism to check `state.status` for `Error` as well and print feedback on success, module already enabled and failure. Please check again ✌️

**Review by jakobmoellersap — APPROVED** [review]

File: internal/cli/alpha/module/module.go
Comment (jakobmoellersap) [inline internal/cli/alpha/module/module.go:76]: If you are running in a timeout, I believe it would be best to output the state and the lastOperation as well as this shows Processing. I also believe the Kyma CR currently incorrectly is set to Processing even though the module is in error state, this has to be fixed in Lifecycle Manager though.

File: docs/gen-docs/kyma_alpha_disable.md
Comment (nataliasitko) [inline docs/gen-docs/kyma_alpha_disable.md:24]:
```suggestion
* [kyma alpha](kyma_alpha.md)	 - Experimental commands.
```

File: docs/gen-docs/kyma_alpha_disable.md
Comment (nataliasitko) [inline docs/gen-docs/kyma_alpha_disable.md:25]:
```suggestion
* [kyma alpha disable module](kyma_alpha_disable_module.md)	 - Disables a module in the cluster or in the given Kyma resource.
```

File: docs/gen-docs/kyma_alpha_enable_module.md
Comment (nataliasitko) [inline docs/gen-docs/kyma_alpha_enable_module.md:5]:
```suggestion
Enables a module in the cluster or in the given Kyma resource.
```

File: docs/gen-docs/kyma_alpha_enable.md
Comment (nataliasitko) [inline docs/gen-docs/kyma_alpha_enable.md:5]:
```suggestion
Enables a resource in the Kyma cluster.
```

File: docs/gen-docs/kyma_alpha_enable.md
Comment (nataliasitko) [inline docs/gen-docs/kyma_alpha_enable.md:9]:
```suggestion
Use this command to enable a resource in the Kyma cluster.
```

File: docs/gen-docs/kyma_alpha_disable.md
Comment (nataliasitko) [inline docs/gen-docs/kyma_alpha_disable.md:5]:
```suggestion
Disables a resource in the Kyma cluster.
```

File: docs/gen-docs/kyma_alpha_disable.md
Comment (nataliasitko) [inline docs/gen-docs/kyma_alpha_disable.md:9]:
```suggestion
Use this command to disable a resource in the Kyma cluster.
```

File: docs/gen-docs/kyma_alpha_disable_module.md
Comment (lindnerby) [reply docs/gen-docs/kyma_alpha_disable_module.md:13]: @nataliasitko Unfortunately the **See Also** section is auto-created by the docs generator and I would not invest time now to create help subcommands to add something extra to this section. Can we keep this as is for now?

Comment (nataliasitko) [reply docs/gen-docs/kyma_alpha_disable_module.md:13]: Yes, we can keep it in this case

File: internal/cli/alpha/module/module.go
Comment (jakobmoellersap) [inline internal/cli/alpha/module/module.go:65]: This will cause further SSAs to result in a conflict. e.g. running `kyma alpha deploy` once, then enabling a module, then running `kyma alpha deploy` again will fail as this is a client side update, not a server side apply. I suggest to

1. Run the enable/disable Patches with Server Side Apply
2. Enable a `force-conflicts` flag similar to `kubectly apply --server-side --force-conflicts` to allow remediation in case a client side apply was issued by another source.

**Review by jakobmoellersap — CHANGES_REQUESTED** [review]

File: internal/cli/alpha/module/module.go
Comment (jakobmoellersap) [inline internal/cli/alpha/module/module.go:74]: is this the same field manager as in the k8s.Apply() call? If so, maybe we should think about using a better field manager like "cli.kyma-project.io". WDYT? Other than that looks good!

File: internal/cli/alpha/module/module.go
Comment (jakobmoellersap) [inline internal/cli/alpha/module/module.go:77]:
```suggestion
	patchOpts.Force = pointer.Bool(i.ForceUpdate)
```

File: internal/cli/command.go
Comment (jakobmoellersap) [inline internal/cli/command.go:33]:
```suggestion
		return fmt.Errorf("failed to discover clusterinfo: %w", err)
```

**Review by jakobmoellersap — COMMENTED** [review]
LGTM :) just waiting for another review

File: internal/cli/alpha/module/module.go
Comment (lindnerby) [reply internal/cli/alpha/module/module.go:74]: Yes, was wondering why manifests are applied with that generic fieldmanager name. guess `kyma-cli` is also ok, compared to other examples like `kubectl-edit` or `manager`?

File: internal/cli/alpha/module/module.go
Comment (lindnerby) [reply internal/cli/alpha/module/module.go]: Definitely should use SSA then. Switched the modules update to a patch with type apply and provided a flag for the forced "resolution".

File: docs/gen-docs/kyma_alpha_disable_module.md
Comment (nataliasitko) [reply docs/gen-docs/kyma_alpha_disable_module.md:34]: I've learned that the shorter the commands' descriptions are, the better, so I removed the unnecessary parts. I'm sorry to bother you again with these.

File: docs/gen-docs/kyma_alpha_enable_module.md
Comment (nataliasitko) [inline docs/gen-docs/kyma_alpha_enable_module.md:36]:
```suggestion
  -f, --force-conflicts    Forces the patching of Kyma spec modules in case their managed field was edited by a source other than Kyma CLI.
```

File: docs/gen-docs/kyma_alpha_enable_module.md
Comment (nataliasitko) [inline docs/gen-docs/kyma_alpha_enable_module.md:35]:
```suggestion
  -c, --channel string     Module's channel to enable.
```

File: docs/gen-docs/kyma_alpha_enable_module.md
Comment (nataliasitko) [inline docs/gen-docs/kyma_alpha_enable_module.md:37]:
```suggestion
  -k, --kyma-name string   Kyma resource to use. If empty, 'default-kyma' is used. (default "default-kyma") (default "default-kyma")
```

File: docs/gen-docs/kyma_alpha_enable_module.md
Comment (nataliasitko) [inline docs/gen-docs/kyma_alpha_enable_module.md:38]:
```suggestion
  -n, --namespace string   Kyma Namespace to use. If empty, the default 'kyma-system' Namespace is used. (default "kyma-system") (default "kyma-system")
```

File: docs/gen-docs/kyma_alpha_list_module.md
Comment (nataliasitko) [inline docs/gen-docs/kyma_alpha_list_module.md:50]:
```suggestion
  -k, --kyma-name string   Kyma resource to use.
```

**Review by ruanxin — DISMISSED** [review]

**Review by nataliasitko — APPROVED** [review]

File: docs/gen-docs/kyma_alpha_disable_module.md
Comment (nataliasitko) [inline docs/gen-docs/kyma_alpha_disable_module.md:36]:
```suggestion
  -k, --kyma-name string   Kyma resource to use. If empty, 'default-kyma' is used. (default "default-kyma")
```

File: docs/gen-docs/kyma_alpha_disable_module.md
Comment (nataliasitko) [inline docs/gen-docs/kyma_alpha_disable_module.md:37]:
```suggestion
  -n, --namespace string   Kyma Namespace to use. If empty, the default 'kyma-system' Namespace is used. (default "kyma-system")
```

File: docs/gen-docs/kyma_alpha_enable_module.md
Comment (nataliasitko) [inline docs/gen-docs/kyma_alpha_enable_module.md:37]:
```suggestion
  -k, --kyma-name string   Kyma resource to use. If empty, 'default-kyma' is used. (default "default-kyma")
```

File: docs/gen-docs/kyma_alpha_enable_module.md
Comment (nataliasitko) [inline docs/gen-docs/kyma_alpha_enable_module.md:38]:
```suggestion
  -n, --namespace string   Kyma Namespace to use. If empty, the default 'kyma-system' Namespace is used. (default "kyma-system")
```

**Review by nataliasitko — APPROVED** [review]

**Review by jakobmoellersap — APPROVED** [review]

**Pull request merged**
